### PR TITLE
[Concept Entry] Luau: iterators.md

### LINKED ISSUE #3208: [Concept Entry] Luau: Iterators
### Reason/inspiration (optional)

We want a new concept entry for [Luau](https://www.codecademy.com/resources/docs/luau) about Iterators.
The file should be located at `content/luau/concepts/iterators/iterators.md`

The entry should:

- Describe what iterators are.
- Have a section the syntax of `while` and `for` loops.
- Give code examples of each type of loop.

Please refer to the [term entry templates](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by noahpgordon**
### Description

Added concept **iterators** to Luau. Closes #3208 

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/luau/concepts/iterators/iterators.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+Output:
+
+```shell
+The value at index 1 is 1
+The value at index 2 is 0.5
+The value at index 5 is biff
+The value at index a is 7
+```
+
+Note that the order in which the elements of the table are looped through is not guaranteed when using `pairs()`.
```
Comment (caupolicandiaz) [inline content/luau/concepts/iterators/iterators.md:161]:
```suggestion
> **Note:** The order in which the elements of the table are looped through is not guaranteed when using `pairs()`.
```

File: content/luau/concepts/iterators/iterators.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+The `ipairs()` function can also be used to iterate through a table, but it has a few key differences from `pairs()`:
+
+1. `ipairs()` will ignore any elements with non-integer indices.
+2. `ipairs()` will begin at index `1` and ignore any elements at earlier indices.
+3. `ipairs()` will stop iterating if there are any integer gaps in the indices.
+4. `ipairs()` guarantees the order in which the elements are looped through.
+
+Due to these differences, `ipairs()` is best used for iterating through arrays.
+
+Compare the behavior of `ipairs()` when used on an identical table as in the previous example:
```
Comment (caupolicandiaz) [inline content/luau/concepts/iterators/iterators.md:174]:
```suggestion
Compare the behavior of `ipairs()` when used on the same table as above:
```

File: content/luau/concepts/iterators/iterators.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+`start_value` initializes the value of `i`. `increment_value` increments the value of `i` after each run of the loop. If `increment_value` is positive, the loop will end once `i` is greater than or equal to `end_value`. If `increment_value` is negative, the will end when `i` is less than or equal to `end_value`.
+
+#### Numeric `for` loop examples
+
+Excluding the optional increment value parameter results in a default increment value of `1`:
+
+```lua
+for i = 1, 3 do
+	print(i)
```
Comment (caupolicandiaz) [inline content/luau/concepts/iterators/iterators.md:102]: be sure to use 2 spaces for code indentations not tabs

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@noahpgordon I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by noahpgordon** [comment]
@caupolicandiaz I've incorporated these edits.

**Review by caupolicandiaz — APPROVED** [review]
@noahpgordon This is ready for a second review.

**Review by SSwiniarski — APPROVED** [review]
@noahpgordon, @caupolicandiaz Looks good to me.

**Pull request merged**